examples/remote-settings-cli/src/downloader.rs

```diff
@@ -0,0 +1,289 @@
+use indicatif::{MultiProgress, ProgressBar, ProgressStyle};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+
+use crate::git_ops::GitOps;
+
+const REMOTE_SETTINGS_URL: &str = "https://firefox.settings.services.mozilla.com/v1";
+const DEFAULT_OUTPUT_DIR: &str = "components/remote_settings";
+const COLLECTIONS: &[&str] = &[
+    "anti-tracking-url-decoration",
+    "cookie-banner-rules-list",
+    "devtools-compatibility-browsers",
+    "devtools-devices",
+    "hijack-blocklists",
+    "language-dictionaries",
+    "password-recipes",
+    "password-rules",
+    "search-config-icons",
+    "search-config-v2",
+    "search-config-overrides-v2",
+    "search-telemetry-v2",
+    "sites-classification",
+    "top-sites",
+    "translations-models",
+    "translations-wasm",
+    "url-classifier-skip-urls",
+    "url-parser-default-unknown-schemes-interventions",
+    "urlbar-persisted-search-terms",
+    "websites-with-shared-credential-backends",
+];
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(crate) struct ServerResponse {
+    pub(crate) data: Vec<Value>,
+    pub(crate) timestamp: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(crate) struct PackagedData {
+    pub(crate) collections: HashMap<String, CollectionData>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub(crate) struct CollectionData {
+    pub(crate) timestamp: u64,
+    pub(crate) records: Vec<Value>,
+}
+
+pub(crate) struct CollectionDownloader {
+    pub(crate) client: reqwest::Client,
+    pub(crate) multi_progress: Arc<MultiProgress>,
+    pub(crate) output_dir: PathBuf,
+}
+
+impl CollectionDownloader {
+    pub(crate) fn new(root_path: PathBuf) -> Self {
+        let output_dir = root_path.join(DEFAULT_OUTPUT_DIR);
+        Self {
+            client: reqwest::Client::new(),
+            multi_progress: Arc::new(MultiProgress::new()),
+            output_dir,
+        }
+    }
+
+    async fn fetch_collection(
+        &self,
+        collection: String,
+        pb: ProgressBar,
+    ) -> Result<(String, CollectionData)> {
+        // First get the last_modified from monitor
+        let monitor_url = format!(
+            "{}/buckets/monitor/collections/changes/records",
+            REMOTE_SETTINGS_URL,
+        );
+
+        let monitor_response: Value = self.client.get(&monitor_url).send().await?.json().await?;
+
+        // Find the last_modified for our collection
+        let last_modified = monitor_response["data"]
+            .as_array()
+            .context("No data array in monitor response")?
+            .iter()
+            .find(|record| {
+                record["collection"].as_str() == Some(&collection)
+                    && record["bucket"].as_str() == Some("main")
+            })
+            .context("Collection not found in monitor")?["last_modified"]
+            .as_u64()
+            .context("No last_modified in record")?;
```

Comment:
I would move this to `download_all()`: fetch the timestamps, and pass them to `fetch_collection()` here by adding a `last_modified` parameter, in order to avoid issuing one request per collection.